src/components/selection_input/selection_input.ts

```diff
@@ -97,19 +103,19 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
   }
 
   get canAddRange(): boolean {
-    return !this.props.hasSingleRange;
+    return !this.props.hasSingleRange && !this.props.readonly;
```

Comment:
Some issues with the readonly style:

- why does the reset/add range button disappear, but the confirm button stays here and is disabled ? We should stay consistent.
- the "trash" button should definitely not be there in readonly
- the selection input should probably lose its colors in readonly. Maybe cancel/confirm the edition to lose focus so we bother lose the colors and the buttons ?
- probably should change the border color rather than the border size when disabling the inputs, otherwise they all jump one pixel up